pipelines/tekton/aiedge-e2e/tasks/model-registry-download.yaml

```diff
@@ -0,0 +1,115 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: model-registry-download
+spec:
+  description: This Task can be used to fetch a model properties from model registry when enabled or properties passed by pipelinerun 
+  params:
+  - name: model-name
+    type: string
+  - name: model-version
+    type: string
+  #Values to be passed by pipeline run when model-register is disabled
+  - name: s3-bucket-name
+    type: string
+    default: ""
+  - name: modelRelativePath
+    type: string
+  - name: model-dir
+    type: string
+    default: "."
+  - name: fetch-model
+    type: string
+  - name: git-model-repo
+    type: string
+    default: ""
+  - name: git-model-revision
+    type: string
+    default: "main"
+  - name: model-registry-hostname
+    type: string
+    default: ""
+  - name: model-registry-enabled
+    type: string
+  results:
+  - name: fetch-model
+    type: string
+    description: The fetch mechanism that will be forwarded to the pipeline
+  - name: git-model-repo
+    description: The GIT URI that will be forwarded to the pipeline
+  - name: s3-bucket-name
+    description: The s3 Bucket name that will be forwarded to the pipeline
+  - name: modelRelativePath
+    type: string
+  - name: model-dir
+    type: string
+  - name: git-model-revision
+    type: string
+  steps:
+  - name: download-model-registry
+    image: registry.access.redhat.com/ubi9/ubi
+    script: |
+      #!/usr/bin/env bash
+
+      if [ "$(params.model-registry-enabled)" == "true" ]; then
+        yum install jq -y
+
+        export MODEL_ID=$(curl --silent -X 'GET' \
+        "$(params.model-registry-hostname)/api/model_registry/v1alpha3/registered_models?" -H 'accept: application/json' | \
```

Comment:
> My thinking was that since the MR would be deployed on the same 'local' cluster as that of the pipeline, so it should/would have access to the endpoint.

That will _probably_ typically be the case, although I do wonder if there will be cases where the pipelines are run on a different cluster than the model registry. I notice that you're using a Route address to access the model registry (rather than a Service address internal to the cluster), which indicates that the model registry will at least be accessible to the outside world.

> I don't think MR have implemented anything with authentication. Do you see anything that I am missing?

Yeah, I'm not sure really. Authentication could be external to the API, using something like the OpenShift OAuth proxy or something else sitting in front of the API.

I'm not sure if it's 100% necessary to be able to handle authentication for this at this point, but it's something that we should at least think about.

@LaVLaS do you have any thoughts about this?

@devguyio have you thought about authentication for the model registry in the CLI at all?

Reply:
@grdryn, I reached out to the MR channel about this question here: https://redhat-internal.slack.com/archives/C05QB0A582E/p1714510019703239. 
I feel the istio part of it is optional so we might be okay temporarily or I might be wrong :)

Reply:
I think it's fine for this PR to leave it as-is, like you say. I think we should have a follow-up story to support authentication on it though. WDYT? If you agree, could you create that in Jira?

Reply:
I agree, other this PR gets too complex and a lot of changes in one. Created the Jira 6703.